### PR TITLE
common returns/throws/requires/emits for polymorphic functions

### LINKED ISSUE #113: shared return type / throws / emits / requires
When you have a polymorphic function and all its signatures share the same return type/throws/emits/requires, it should be written only once. Obviously only allowed if _everything_ is shared.

e.g.

```
interface Options {
  capture: Boolean,
  once: Boolean,
  passive Boolean,
  mozSystemGroup: Boolean
}

interface addEventListener {
  (type: String, listener: Function | Event, Options?),
  (type: String, listener: Function | Event, useCapture: Boolean),
  (type: String, listener: Function | Event, useCapture: Boolean, wantsUntrusted?: Boolean)
} => Void




/*
} => Type,
  throws: ErrorType,
  requires: example,
  emits: { … }
*/
```

### PR DESCRIPTION
**Pull request by Mouvedia**
close #113

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -400,6 +400,16 @@ interface Collection {
 }
 ```
 
+If all signatures return/emit/throw/require the same thing, you can consolidate this information in one place:
+
+```js
+interface addEventListener {
+  (type: String, listener: Function | Event, Options?), // assumes Options has been declared before
```
Comment (ericelliott) [inline README.md]: Let's simplify this example. The point is not in the details of these various interfaces, but in the shared `=> void` return type, right? Let's clarify that.

**Review by ericelliott — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (Mouvedia) [reply README.md]: I removed the Options interface for that specific reason. I should have gone even further.

**Comment by Mouvedia** [comment]
@ericelliott looks ready to me.

**Review by ericelliott — APPROVED** [review]
Much better!

**Pull request merged**